src/Symfony/Component/Lock/Store/PdoStore.php

```diff
@@ -0,0 +1,373 @@
+        }
+
+        switch ($driver) {
+            case 'mysql':
+                $sql = "CREATE TABLE $this->table ($this->idCol VARCHAR(64) NOT NULL PRIMARY KEY, $this->tokenCol VARCHAR(44) NOT NULL, $this->expirationCol INTEGER UNSIGNED NOT NULL) COLLATE utf8_bin, ENGINE = InnoDB";
+                break;
+            case 'sqlite':
+                $sql = "CREATE TABLE $this->table ($this->idCol TEXT NOT NULL PRIMARY KEY, $this->tokenCol TEXT NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            case 'pgsql':
+                $sql = "CREATE TABLE $this->table ($this->idCol VARCHAR(64) NOT NULL PRIMARY KEY, $this->tokenCol VARCHAR(64) NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            case 'oci':
+                $sql = "CREATE TABLE $this->table ($this->idCol VARCHAR2(64) NOT NULL PRIMARY KEY, $this->tokenCol VARCHAR2(64) NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            case 'sqlsrv':
+                $sql = "CREATE TABLE $this->table ($this->idCol VARCHAR(64) NOT NULL PRIMARY KEY, $this->tokenCol VARCHAR(64) NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            default:
+                throw new \DomainException(sprintf('Creating the lock table is currently not implemented for PDO driver "%s".', $driver));
+        }
+
+        $conn->exec($sql);
+    }
+
+    /**
+     * Cleanup the table by removing all expired locks
+     */
+    private function prune(): void
+    {
+        $sql = "DELETE FROM $this->table WHERE $this->expirationCol <= {$this->getCurrentTimestampStatment()}";
+
+        $stmt = $this->getConnection()->prepare($sql);
+
+        $stmt->execute();
+    }
+
+    private function getDriver(): string
+    {
+        if (null !== $this->driver) {
+            return $this->driver;
+        }
+
+        $con = $this->getConnection();
+        if ($con instanceof \PDO) {
+            $this->driver = $con->getAttribute(\PDO::ATTR_DRIVER_NAME);
+        } else {
+            switch ($this->driver = $con->getDriver()->getName()) {
+                case 'mysqli':
+                case 'pdo_mysql':
+                case 'drizzle_pdo_mysql':
+                    $this->driver = 'mysql';
+                    break;
+                case 'pdo_sqlite':
+                    $this->driver = 'sqlite';
+                    break;
+                case 'pdo_pgsql':
+                    $this->driver = 'pgsql';
+                    break;
+                case 'oci8':
+                case 'pdo_oracle':
+                    $this->driver = 'oci';
+                    break;
+                case 'pdo_sqlsrv':
+                    $this->driver = 'sqlsrv';
+                    break;
+            }
+        }
+
+        return $this->driver;
+    }
+
+
+    private function getCurrentTimestampStatment(): string
+    {
+        switch ($this->getDriver()) {
+            case 'mysql':
+                return 'UNIX_TIMESTAMP()';
```

Comment:
If we don't want to be sensible to seasonal time changes, we have to use UTC.

Reply:
Unix timestamp should not be timezoned, it's always the amount of seconds since 1970-01-01 00:00:00 UTC

Reply:
> Some functions are affected by the time zone settings. These include `UNIX_TIMESTAMP()`
https://mariadb.com/kb/en/library/time-zones/#time-zone-effects

Reply:
They say the opposite in https://mariadb.com/kb/en/library/unix_timestamp/ :/
> The server interprets date as a value in the current time zone and converts it to an internal value in UTC.

Point is, the function `UTC_TIMESTAMP` does not returns an unix timestamp (integer represetntation of seconds since 1970-01-01), but it returns the current date with UTC timezone, this method is equivalent to `SET time_zone = 'UTC'; SELECT NOW();` 

Just test on MySQL 5 (same results with latest mariadb).
As far as I understand: UNIX_TIMESTAMP is what we want: an invariable and linear (as far as the server's clock is not changed) time representation.
```
SET time_zone = 'UTC';select UNIX_TIMESTAMP();SET time_zone = 'europe/paris';select UNIX_TIMESTAMP();
Query OK, 0 rows affected (0.00 sec)

+------------------+
| UNIX_TIMESTAMP() |
+------------------+
|       1528274441 |
+------------------+
1 row in set (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

+------------------+
| UNIX_TIMESTAMP() |
+------------------+
|       1528274441 |
+------------------+
1 row in set (0.01 sec)
```
The timestamp is identical, wheras with `NOW()`
```
SET time_zone = 'UTC';select NOW();SET time_zone = 'europe/paris';;select NOW();
Query OK, 0 rows affected (0.00 sec)

+---------------------+
| NOW()               |
+---------------------+
| 2018-06-06 08:43:16 |
+---------------------+
1 row in set (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

+---------------------+
| NOW()               |
+---------------------+
| 2018-06-06 10:43:16 |
+---------------------+
1 row in set (0.00 sec)
```